utt/libutt/include/utt/DataUtils.hpp

```diff
@@ -0,0 +1,65 @@
+#pragma once
+
+#include <utt/IBE.h>
+
+#include <openssl/rsa.h>
+#include <openssl/pem.h>
+#include <openssl/x509.h>
+#include <openssl/bio.h>
+#include <openssl/evp.h>
+
+#include <vector>
+#include <cstdint>
+#include <unordered_map>
+#include <memory>
+namespace libutt {
+class IEncryptor {
+ public:
+  virtual std::vector<uint8_t> encrypt(const std::string& id, const std::vector<uint8_t>& msg) const = 0;
+  virtual ~IEncryptor() {}
+};
+
+class IDecryptor {
+ public:
+  virtual std::vector<uint8_t> decrypt(const std::vector<uint8_t>&) const = 0;
+  virtual ~IDecryptor() {}
+};
+
+class IBEEncryptor : public IEncryptor {
```

Comment:
What does IBE stand for?

Reply:
IBE = identity based encryption. Please see the UTT paper for reference

Reply:
If this is important, it should probably be explained in a comment..
Unless it is a requirement to read the paper before making any modification to this file, which should also be specified somehow in the source so that people who didn't will stay away.